### PR TITLE
ARCH-1615 - Actions maintenance

### PR DESCRIPTION
**Pull request by danielle-casella-adams**
# Summary of PR changes

- Only include certain files when considering when to increment the tag.
- Remove the restriction on the build workflow, so we always get a status check
- Update actions to their latest tags
- This action already had node updated, so it will only have a patch update
- Update actions/core to v1.10.0 to handle [untrusted logged data](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/)

## PR Requirements
- [x] For JavaScript actions, the action has been recompiled.  
  - See the *Recompiling* section of the repository's README.md for more details.
  - This does not apply to Composite Run Steps actions unless a package.json is present and contains a build script.
- [x] For major, minor, or breaking changes, at least one of the commit messages contains the appropriate `+semver:` keywords.
  - See the *Incrementing the Version* section of the repository's README.md for more details.
- [x] The examples in the repository's `README.md` have been updated with the new version.  
  - See the *Incrementing the Version* section of the repository's README.md for more details on how the version will be incremented.
- [x] The action code does not contain sensitive information.

### CODE REVIEW

**Review by JosephDSchwartz — APPROVED** [review]

**Review by JosephDSchwartz — COMMENTED** [review]
It might make sense to include the changes from https://github.com/im-open/process-dotnet-test-results/pull/19 in this pull request. Less version updates and all that.

**Comment by danielle-casella-adams** [comment]
I merged @ridomin's PR #19 in so this one should be good to go.

**Review by JosephDSchwartz — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -95,15 +95,15 @@ jobs:
...
+      - uses: actions/checkout@v3
 
       - name: Test My Solution
         continue-on-error: true
         run: dotnet test ./src/my-solution.sln --logger "trx" --configuration Release --results-directory ../../test-results
       
       - name: Process trx reports
         id: process-trx
-        uses: im-open/process-dotnet-test-results@v2.2.0
+        uses: im-open/process-dotnet-test-results@v2.2.1
```
Comment (JosephDSchwartz) [inline README.md:106]: This will be 2.2.2 now right?

Comment (danielle-casella-adams) [reply README.md:106]: Oh man I thought I was being clever...but this one won't have a minor increment.  Shoot!

File: README.md [hunk 2]
```diff
@@ -77,14 +77,14 @@ jobs:
...
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
       - name: dotnet test with coverage
         run: dotnet test './src/MyProj.sln' --logger trx --configuration Release
 
       - name: Process trx reports with default
         if: always()
-        uses: im-open/process-dotnet-test-results@v2.2.0
+        uses: im-open/process-dotnet-test-results@v2.2.1
```
Comment (danielle-casella-adams) [inline README.md:87]:
```suggestion
        uses: im-open/process-dotnet-test-results@v2.2.2
```

File: README.md (hunk 1 above)
Comment (danielle-casella-adams) [inline README.md:106]:
```suggestion
        uses: im-open/process-dotnet-test-results@v2.2.2
```

Comment (danielle-casella-adams) [reply README.md]: The updated versions are pushed.  Sorry about that confusion.  I'm confused. :)

**Review by JosephDSchwartz — APPROVED** [review]

**Pull request merged**